public/app/features/query/components/QueryEditorRows.tsx

```diff
@@ -128,9 +135,68 @@ export class QueryEditorRows extends PureComponent<Props> {
     });
   };
 
+  onAddCondition = (queryIdx: number) => (conditionId: QueryConditionID) => {
+    const { queries, onQueriesChange } = this.props;
+    const conditionDef = queryConditionsRegistry.getIfExists(conditionId);
+
+    if (!conditionDef) {
+      console.error(`Could not find condition definition for ${conditionId}`);
+    }
+
+    // eslint-disable-next-line
+    const q = queries[queryIdx] as ConditionalDataSourceQuery;
+    q.conditions = q.conditions
+      ? [...q.conditions, { id: conditionId, options: { ...conditionDef!.defaultOptions } }]
+      : [{ id: conditionId, options: { ...conditionDef!.defaultOptions } }];
+
+    const nextQueries = [...queries];
+    nextQueries[queryIdx] = q;
+
+    onQueriesChange(nextQueries);
+  };
+
+  onRemoveCondition = (queryIdx: number) => (idx: number) => {
+    const { queries, onQueriesChange } = this.props;
+
+    // eslint-disable-next-line
+    const q = queries[queryIdx] as ConditionalDataSourceQuery;
+    q.conditions.splice(idx, 1);
+    const nextQueries = [...queries];
+    nextQueries[queryIdx] = q;
+
+    onQueriesChange(nextQueries);
+  };
+
+  // eslint-disable-next-line
+  onQueryConditionChange = (queryIdx: number) => (conditionIdx: number, options: any) => {
+    const { queries, onQueriesChange } = this.props;
+
+    // eslint-disable-next-line
+    const q = queries[queryIdx] as ConditionalDataSourceQuery;
+    q.conditions[conditionIdx].options = options;
+
+    const nextQueries = [...queries];
+    nextQueries[queryIdx] = q;
+
+    onQueriesChange(nextQueries);
+  };
+
+  renderConditionsEditor(q: ConditionalDataSourceQuery, queryIdx: number) {
+    return (
+      <QueryConditionsEditor
+        conditions={q.conditions}
+        onChange={this.onQueryConditionChange(queryIdx)}
+        onAddCondition={this.onAddCondition(queryIdx)}
+        onRemoveCondition={this.onRemoveCondition(queryIdx)}
+      />
+    );
+  }
```

Comment:
Yes, totally. I was thinking about way of utilising conditional data source's query editor, but got stuck on that one, mainly because of the mixed data source nature and the way we currently render those editors.